packages/intrepid2/src/Discretization/Basis/Intrepid2_Basis.hpp

```diff
@@ -75,6 +75,11 @@ class Basis;
 template <typename DeviceType = void, typename OutputType = double, typename PointType = double>
 using BasisPtr = Teuchos::RCP<Basis<DeviceType,OutputType,PointType> >;
 
+/** \brief Pointer to a Basis whose device type is on the host (Kokkos::HostSpace::device_type), allowing host access to input and output views, and ensuring host execution of basis evaluation.
+ */
+template <typename OutputType = double, typename PointType = double>
+using HostBasisPtr = BasisPtr<typename Kokkos::HostSpace::device_type, OutputType, PointType>;
```

Comment:
Why do you do defaulting for the output type and point type ? This should match to this output type and point type. It does not need to default with double.

Reply:
@kyungjoo-kim, I followed the pattern of the BasisPtr typedef that is immediately above this typedef.  It's true it doesn't need to default to double, but I think it's good to be consistent between the two typedefs.

Reply:
This is internal typedef of the basis. Defaulting the template arguments here and in the above does not make sense to me.

Reply:
@kyungjoo-kim, ah, I see the confusion.  Actually, these typedefs are not internal to the Basis class.